File: collect_app/src/main/java/org/odk/collect/android/formmanagement/FormDownloadException.java
package org.odk.collect.android.formmanagement;

public class FormDownloadException extends Exception {
    public FormDownloadException() {
        super();
    }

    public FormDownloadException(String message) {
        super(message);
    }

    public static class DownloadingInterruptedException extends FormDownloadException {

    }

    public static class FormWithNoHashException extends FormDownloadException {

    }

    public static class FormParsingException extends FormDownloadException {

    }

    public static class DiskException extends FormDownloadException {

    }
}

File: collect_app/src/main/java/org/odk/collect/android/formmanagement/FormDownloadException.kt
package org.odk.collect.android.formmanagement

import org.odk.collect.forms.FormSourceException

sealed class FormDownloadException : Exception() {
    class DownloadingInterrupted : FormDownloadException()
    class FormWithNoHash : FormDownloadException()
    class FormParsingError : FormDownloadException()
    class DiskError : FormDownloadException()
    class InvalidSubmission : FormDownloadException()
    class FormSourceError(val exception: FormSourceException) : FormDownloadException()
}
